luigi/parameter.py

```diff
@@ -219,7 +219,14 @@ def parse(self, x):
         :param str x: the value to parse.
         :return: the parsed value.
         """
-        return x  # default impl
+        try:
+            if not isinstance(x, basestring):   # python 2
+                warnings.warn("Parameter %r is not of type string." % x)
+        except:
+            if not isinstance(x, str):          # python 3
+                warnings.warn("Parameter %r is not of type string." % x)
+        finally:
```

Comment:
However, i'm open to suggestion on how to clean this up and not have the `warnings.warn(...)` message written twice.

The only other thing i can think of to only write the message once is:
```python
warn = False
try:
    if not isinstance(x, basestring):
        warn = True
except:
    if not isinstance(x, str):
        warn = True
finally:
    if warn:
        warnings.warn("Parameter %r is not of type string." % x)
    return x
```